### PR TITLE
METRON-2222 Remove Overrides for Storm 1.0.x

### PR DESCRIPTION
**Pull request by nickwallen**
The `metron-storm-kafka-overrides` project contains some overridden `storm-kafka` classes that were necessary to address a performance issues in older versions of `storm-kafka`.  Now that we have upgraded to Storm 1.2.1 these fixes are included in the code base and the overrides are no longer needed. 

I have validated that the version of `storm-kafka` we now build against matches the overrides we had in place for the following classes.
* org/apache/storm/kafka/spout/KafkaSpoutRetryExponentialBackoff.java
* org/apache/storm/kafka/spout/internal/Timer.java

## Acceptance Testing

This PR should be tested using the centos7 development environment.  

1. Start up the centos7 dev environment.
    ```
    cd metron-deployment/development/centos7
    vagrant destroy -f
    vagrant up
    ```

### Basic Indexing and Enrichment

Ensure that we can continue to parse, enrich, and index telemetry.  Verify data is flowing through the system, from parsing to indexing

1. Open Ambari and navigate to the Metron service http://node1:8080/#/main/services/METRON/summary

1. Open the Alerts UI.  Verify alerts show up in the main UI - click the search icon (you may need to wait a moment for them to appear)

1. Go to the Alerts UI and ensure that an ever increasing number of telemetry from Bro, Snort, and YAF are visible by watching the total alert count increase over time.

1. Ensure that geoip enrichment is occurring.  The telemetry should contain fields like `enrichments:geo:ip_src_addr:location_point`.

1. Head back to Ambari and select the Kibana service http://node1:8080/#/main/services/KIBANA/summary

1. Open the Kibana dashboard via the "Metron UI" option in the quick links

1. Verify the dashboard is populating

## Pull Request Checklist

- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### CODE REVIEW

File: metron-platform/metron-storm-kafka/pom.xml [hunk 1]
```diff
@@ -77,11 +84,89 @@
...
+                          <relocations>
+                                <relocation>
+                                    <pattern>com.google.common</pattern>
+                                    <shadedPattern>org.apache.metron.guava.${guava_version}</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>com.google.thirdparty</pattern>
+                                    <shadedPattern>org.apache.metron.storm.kafka.override.guava.thirdparty</shadedPattern>
+                                </relocation>
+                                <!-- <relocation>
```
Comment (mmiklavc) [inline metron-platform/metron-storm-kafka/pom.xml]: Should this commented out bit be removed?

Comment (nickwallen) [reply metron-platform/metron-storm-kafka/pom.xml]: Oops, yes.  I cleaned this up.

File: metron-platform/metron-storm-kafka/pom.xml [hunk 2]
```diff
@@ -77,11 +84,81 @@
...
+                            </filter>
+                          </filters>
+                          <relocations>
+                                <relocation>
+                                    <pattern>com.google.common</pattern>
+                                    <shadedPattern>org.apache.metron.guava.${guava_version}</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>com.google.thirdparty</pattern>
+                                    <shadedPattern>org.apache.metron.storm.kafka.override.guava.thirdparty</shadedPattern>
```
Comment (mmiklavc) [inline metron-platform/metron-storm-kafka/pom.xml]: Should we name this consistently with com.google.common? i.e. `org.apache.metron.com.google.thirdparty`?

**Comment by mmiklavc** [comment]
Other than the relocation nit, this looks good @nickwallen. Love cleaning up code branches! +1 by inspection once that's done.

File: metron-platform/metron-storm-kafka/pom.xml (hunk 2 above)
Comment (nickwallen) [reply metron-platform/metron-storm-kafka/pom.xml]: Yeah, I can clean that up.  I had just copied that out directly from metron-storm-kafka-override, but I should make sure the pattern matches the other places that we do this.

**Comment by nickwallen** [comment]
Thanks @mmiklavc . 

This has been merged into `feature/METRON-2088-support-hdp-3.1`

**Pull request closed without merging**